### DISCUSSION TITLE
Just SSR per every request

### DISCUSSION THREAD

**Discussion by zvitek** [vercel/next.js#52128] [Help]
### Summary

Hey, guys,
is it possible to use NextJS only for server render? So that each click on the link again executes the whole request/response without SPA.

Ideally, so that it doesn't even create a client build of the application.

Thank you

**Comment by icyJoseph** [comment]
One cheap, quick and dirty way would be, to use `a` tags instead of Link so that the router is not even used. 

This would happen naturally with JavaScript disabled. Next.js is ready for that.

Then you can just pre-render the app at build time, or on demand via various mechanisms.

---

It feels like there's two questions at once here. 

What do you mean with SSR here? SSR is a term that's thrown around a lot, but do you mean, pre-render at build time, or render on-demand? 

I think you want on-demand, for that you'd just need to `export const dynamic = "force-dynamic"` at page level. Your build a bunch of  `λ` pages. 

> λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)

Though there's an issue with the client side cache of Server Components. Hopefully that's not an issue in your case, so that transitioning back to pages already generated, might show stale data. Server Actions can help with this.

**Reply by zvitek** [reply]
Thanks for the reply, we are resolving that we would like to use react as a template only.
On input we have a JSON structure that we want to send to the application and then, the whole HTML to send to response.

For each request, a new page will be built this way.
